### PR TITLE
base & kraken - common currencies

### PR DESCRIPTION
**Pull request by ttodua**

### CODE REVIEW

File: ts/src/base/Exchange.ts [hunk 1]
```diff
@@ -5049,13 +5049,6 @@ export default class Exchange {
         if (!this.substituteCommonCurrencyCodes) {
             return code;
         }
-        // if the provided code already exists as a value in commonCurrencies dict, then we should not again transform it
-        // more details at: https://github.com/ccxt/ccxt/issues/21112#issuecomment-2031293691
-        const commonCurrencies = Object.values (this.commonCurrencies);
-        const exists = this.inArray (code, commonCurrencies);
-        if (exists) {
-            return code;
-        }
```
Comment (ttodua) [inline ts/src/base/Exchange.ts:5058]: reverted this changed (which was merged few days ago)

File: ts/src/base/Exchange.ts [hunk 2]
```diff
@@ -5746,7 +5739,6 @@ export default class Exchange {
          * @returns {object} objects with withdraw and deposit fees, indexed by currency codes
          */
         const depositWithdrawFees = {};
-        codes = this.marketCodes (codes);
```
Comment (ttodua) [inline ts/src/base/Exchange.ts:5749]: no need this method to be present here, in request side

File: ts/src/kraken.ts [hunk 3]
```diff
@@ -669,15 +674,20 @@ export default class kraken extends Exchange {
...
+            const altname = this.safeString (currency, 'altname');
+            if (altname !== id) {
+                // only allow for them which has different altname (XBT) and id (XXBT)
+                this.options['sliceAllowedForCurrencyIds'].push (id);
+            }
+            const code = this.safeCurrencyCode (id);
             const precision = this.parseNumber (this.parsePrecision (this.safeString (currency, 'decimals')));
             // assumes all currencies are active except those listed above
-            const active = !this.inArray (code, this.options['inactiveCurrencies']);
+            const active = this.safeString (currency, 'status') === 'enabled';
```
Comment (ttodua) [inline ts/src/kraken.ts:679]: they added new field in api response, so we can directly use this instead of hardcoded list

File: ts/src/kraken.ts [hunk 4]
```diff
@@ -231,6 +232,7 @@ export default class kraken extends Exchange {
                 'delistedMarketsById': {},
                 // cannot withdraw/deposit these
                 'inactiveCurrencies': [ 'CAD', 'USD', 'JPY', 'GBP' ],
+                'sliceAllowedForCurrencyIds': [],
```
Comment (carlosmiei) [inline ts/src/kraken.ts]: @ttodua what's this?

Comment (ttodua) [reply ts/src/kraken.ts]: ah, remnant from previous commit, sry, removing it 1 sec

File: ts/src/base/Exchange.ts (hunk 1 above)
Comment (sc0Vu) [reply ts/src/base/Exchange.ts:5058]: @ttodua LGTM, but how do we solve issue https://github.com/ccxt/ccxt/issues/21112 ?

Comment (ttodua) [reply ts/src/base/Exchange.ts:5058]: > @ttodua LGTM, but how do we solve issue [ccxt#21112](https://github.com/ccxt/ccxt/issues/21112) ?

so, removal of `codes = this.marketCodes (codes);` from fDWF method will fix that problem (as that OP was experiencing issue specifically with fDWF method).

**Pull request closed without merging**